mycore-base/src/main/java/org/mycore/datamodel/common/MCRFileBaseCacheObjectIDGenerator.java

```diff
@@ -50,36 +54,40 @@ public class MCRFileBaseCacheObjectIDGenerator implements MCRObjectIDGenerator {
     static ConcurrentHashMap<String, ReentrantReadWriteLock> locks = new ConcurrentHashMap<>();
 
     private static Path getCacheFilePath(String baseId) {
+        Path idCachePath = getCacheDirPath();
 
-        Path dataDir = getDataDirPath();
-
-        Path idCachePath = dataDir.resolve("id_cache");
-        if (!Files.exists(idCachePath)) {
+        Path cacheFile = MCRUtils.safeResolve(idCachePath, baseId);
+        if (!Files.exists(cacheFile)) {
```

Comment:
Avoid nesting:
```
Path cacheFile = MCRUtils.safeResolve(idCachePath, baseId);
if(Files.exists(cacheFile)) {
  return cacheFile;
}
...
```
Looks cleaner.

Reply:
Fixed in https://github.com/MyCoRe-Org/mycore/pull/2095